kodilla-patterns/src/test/java/com/koilla/patterns/prototype/library/LibraryTestSuite.java

```diff
@@ -0,0 +1,82 @@
+package com.koilla.patterns.prototype.library;
+
+import com.kodilla.patterns.prototype.library.Book;
+import com.kodilla.patterns.prototype.library.Library;
+import org.junit.Assert;
+import org.junit.Test;
+import java.time.LocalDate;
+import java.util.stream.IntStream;
+
+public class LibraryTestSuite {
+
+    @Test
+    public void testGetBooks() {
+        //Given
+        Library library = new Library("Library of Alexandria");
+        IntStream.iterate(0, n -> n + 1 )
+                .limit(9)
+                .forEach(n -> library.getBooks().add(new Book("Ulysses" + n, "James Joyce" + n, LocalDate.of(2020 - n, 10 - n, 30 - n))));
+        //When
+        Library libraryTest = new Library("Library of Alexandria");
+        //Then
+        Assert.assertEquals(libraryTest.getName(), library.getName());
+        Assert.assertEquals(9, library.getBooks().size());
+    }
+
+    @Test
+    public void testGetBooksShallowCopy() {
+        //Given
+        Library library = new Library("Library of Alexandria");
+        IntStream.iterate(0, n -> n + 1 )
```

Comment:
To zdaje się niepotrzebnie skomplikowane. Aby dobrze przetestować oba rodzaje klonowania, wystarczą Ci dwa elementy (i usunięcie jednego w testach), albo odwrotnie - jeden element na start i później dodany drugi. Nie potrzeba 10-ciu.